cli/daemon/run/run_test.go

```diff
@@ -88,211 +97,235 @@ func TestStartProc(t *testing.T) {
+			run.ServeHTTP(w, req)
+			c.Assert(w.Code, qt.Equals, 200)
+		}
 
-	// Call an endpoint with request parameters but no response value
-	{
-		w := httptest.NewRecorder()
-		req := httptest.NewRequest("GET", "/echo.MuteEcho?key=pelican&value=cocabura", nil)
-		run.ServeHTTP(w, req)
-		c.Assert(w.Code, qt.Equals, 200)
-	}
+		// Call an endpoint with request parameters but no response value
+		{
+			w := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/echo.MuteEcho?key=pelican&value=cocabura", nil)
+			run.ServeHTTP(w, req)
+			c.Assert(w.Code, qt.Equals, 200)
+		}
 
-	// Call an endpoint with a response value but no request parameters
-	{
-		w := httptest.NewRecorder()
-		req := httptest.NewRequest("GET", "/echo.Pong", nil)
-		run.ServeHTTP(w, req)
-		c.Assert(w.Code, qt.Equals, 200)
-		c.Assert(w.Body.Bytes(), qt.JSONEquals, Data[string, string]{"woodpecker", "kingfisher"})
-	}
+		// Call an endpoint with a response value but no request parameters
+		{
+			w := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/echo.Pong", nil)
+			run.ServeHTTP(w, req)
+			c.Assert(w.Code, qt.Equals, 200)
+			c.Assert(w.Body.Bytes(), qt.JSONEquals, Data[string, string]{"woodpecker", "kingfisher"})
+		}
 
-	// Call the env endpoint and make sure we get our env variables back
-	{
-		w := httptest.NewRecorder()
-		req := httptest.NewRequest("GET", "/echo.Env", nil)
-		run.ServeHTTP(w, req)
-		c.Assert(w.Code, qt.Equals, 200)
-		c.Assert(w.Body.Bytes(), qt.JSONEquals, map[string][]string{"Env": wantEnv})
-	}
+		// Call the env endpoint and make sure we get our env variables back
+		{
+			w := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/echo.Env", nil)
+			run.ServeHTTP(w, req)
+			c.Assert(w.Code, qt.Equals, 200)
+			c.Assert(w.Body.Bytes(), qt.JSONEquals, map[string][]string{"Env": wantEnv})
+		}
 
-	// Call the app metadata endpoint and make sure we get correct data back
-	{
-		w := httptest.NewRecorder()
-		req := httptest.NewRequest("GET", "/echo.AppMeta", nil)
-		run.ServeHTTP(w, req)
-		c.Assert(w.Code, qt.Equals, 200)
-
-		// we need to extract the API Base URL as it will change due to the `RuntimePort: 0` above
-		bytes := w.Body.Bytes()
-		got := make(map[string]string)
-		_ = json.Unmarshal(bytes, &got)
-
-		c.Assert(strings.HasPrefix(got["APIBaseURL"], "http://"), qt.IsTrue)
-		c.Assert(bytes, qt.JSONEquals, map[string]interface{}{
-			"AppID":      "slug",
-			"APIBaseURL": got["APIBaseURL"],
-			"EnvName":    "local",
-			"EnvType":    "local",
-		})
-	}
+		// Call the app metadata endpoint and make sure we get correct data back
+		{
+			w := httptest.NewRecorder()
+			req := httptest.NewRequest("GET", "/echo.AppMeta", nil)
+			run.ServeHTTP(w, req)
+			c.Assert(w.Code, qt.Equals, 200)
+
+			// we need to extract the API Base URL as it will change due to the `RuntimePort: 0` above
+			bytes := w.Body.Bytes()
+			got := make(map[string]string)
+			_ = json.Unmarshal(bytes, &got)
```

Comment:
Is it guaranteed all values are strings or should this be `map[string]any`?

Reply:
🤷 This was the original test code, I've not modified it apart from tabbing it in to run as a sub test.

Reply:
Yes it is, this is the response type in the test app

```
type AppMetadata struct {
	AppID      string
	APIBaseURL string
	EnvName    string
	EnvType    string
}
```